apps/remote_control/lib/lexical/remote_control/code_mod/ast/aliases.ex

```diff
@@ -0,0 +1,300 @@
+    defp apply_ast(%__MODULE__{} = reducer, {_definition, metadata, _body}) do
+      if Keyword.has_key?(metadata, :end) do
+        push_scope(reducer, metadata, current_module(reducer))
+      else
+        reducer
+      end
+    end
+
+    defp apply_ast(%__MODULE__{} = reducer, _elem) do
+      reducer
+    end
+
+    defp apply_multiple_aliases(%__MODULE__{} = reducer, from_alias, destinations) do
+      Enum.reduce(destinations, reducer, fn {:__aliases__, _, to_alias}, reducer ->
+        from =
+          case from_alias do
+            [:__MODULE__ | rest] ->
+              [:__MODULE__ | rest ++ to_alias]
+
+            from ->
+              from ++ to_alias
+          end
+
+        to = List.last(from)
+        put_alias(reducer, from, to)
+      end)
+    end
+
+    defp put_alias(%__MODULE__{} = reducer, _, :skip) do
+      reducer
+    end
+
+    defp put_alias(%__MODULE__{} = reducer, from, to) do
+      scope =
+        reducer
+        |> current_scope()
+        |> Scope.put_alias(from, to)
+
+      replace_current_scope(reducer, scope)
+    end
+
+    defp current_scope(%__MODULE__{scopes: [current | _]}) do
+      current
+    end
+
+    defp replace_current_scope(%__MODULE__{scopes: [_ | rest]} = reducer, scope) do
+      %__MODULE__{reducer | scopes: [scope | rest]}
+    end
+
+    defp ensure_alias(%__MODULE__{} = reducer, [{:__MODULE__, _, _}, rest]) do
+      reducer
+      |> current_module()
+      |> Module.concat(rest)
+    end
+
+    defp ensure_alias(%__MODULE__{}, alias_list) when is_list(alias_list) do
+      Module.concat(alias_list)
+    end
+
+    defp ensure_alias(%__MODULE__{}, alias_atom) when is_atom(alias_atom) do
+      alias_atom
+    end
+
+    defp push_scope(
+           %__MODULE__{} = reducer,
+           metadata,
+           current_module,
+           on_exit \\ &Function.identity/1
+         ) do
+      end_position = {get_in(metadata, [:end, :line]), get_in(metadata, [:end, :column])}
+      current_module = ensure_alias(reducer, current_module)
+      new_scopes = [Scope.new(end_position, current_module, on_exit) | reducer.scopes]
+
+      %__MODULE__{reducer | scopes: new_scopes}
+    end
+
+    defp maybe_pop_scope(%__MODULE__{} = reducer, {_, metadata, _}) do
+      with {:ok, current_line} <- Keyword.fetch(metadata, :line),
+           {:ok, current_column} <- Keyword.fetch(metadata, :column),
+           [current_scope | scopes] <- reducer.scopes,
+           true <- Scope.ended?(current_scope, {current_line, current_column}) do
+        current_scope.on_exit.(%__MODULE__{reducer | scopes: scopes})
+      else
+        _ ->
+          reducer
+      end
+    end
+
+    defp maybe_pop_scope(%__MODULE__{} = reducer, _) do
+      reducer
+    end
+
+    defp normalize_from([{:__MODULE__, _, _} | rest]) do
+      [:__MODULE__ | rest]
+    end
+
+    defp normalize_from(from) do
+      from
+    end
+  end
+
+  alias Future.Code
```

Comment:
which version of elixir will this be released as eventually?

Reply:
This is because José made changes for us, so we vendored the 1.5 `Code` and associated modules and namespaced them in the `Future` module. The nice thing about this is we can just `alias Future.Code` and you have a `Code` module.... from the future.